File: kotlin/Hash Maps and Sets/PairSumUnsorted.kt
fun pairSumUnsorted(nums: List<Int>, target: Int): List<Int> {
    val hashmap = mutableMapOf<Int, Int>()
    for ((i, x) in nums.withIndex()) {
        if (target - x in hashmap) {
            return listOf(hashmap[target - x]!!, i)
        }
        hashmap[x] = i
    }
    return emptyList()
}

File: typescript/Hash Maps and Sets/pair_sum_unsorted.ts
function pairSumUnsorted(nums: number[], target: number): number[] {
    const hashMap: { [key: number]: number } = {};
    for (let i = 0; i < nums.length; i++) {
        if (hashMap[target - nums[i]] !== undefined)
            return [hashMap[target - nums[i]], i];
        hashMap[nums[i]] = i;
    }
    return [];
}
